shared/src/amm_pair_provider.rs

```diff
@@ -1,12 +1,15 @@
-use contracts::UniswapV2Factory;
+use contracts::{SushiswapV2Factory, UniswapV2Factory};
 use ethcontract::H160;
 use hex_literal::hex;
 use model::TokenPair;
 use web3::signing::keccak256;
+
 const UNISWAP_PAIR_INIT_CODE: [u8; 32] =
     hex!("96e8ac4277198ff8b6f785478aa9a39f403cb768dd02cbee326c3e7da348845f");
 const HONEYSWAP_PAIR_INIT_CODE: [u8; 32] =
     hex!("3f88503e8580ab941773b59034fb4b2a63e86dbc031b3633a925533ad3ed2b93");
+const SUSHI_PAIR_INIT_CODE: [u8; 32] =
+    hex!("e18a34eb0e04b04f7a0ac29a6e80748dca96319b42c54d679cb821dca90c6303");
```

Comment:
For which networks is this init code valid?

Reply:
All networks

Reply:
I'll include a test to demonstrate on rinkeby and xdai.